modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/TestData/Sources/ExportedProperties2.cs

```diff
@@ -0,0 +1,16 @@
+using Godot;
+using System;
+
+[GlobalClass]
+public partial class ExportedProperties2(int health, Resource subResource, string[] strings) : Resource
+{
+    [Export]
+    public int Health { get; set; } = health;
+
+    public Resource SubResource { get; set; } = subResource;
+
+    public string[] Strings { get; set; } = strings ?? System.Array.Empty<string>();
+
+    // Make sure you provide a parameterless constructor.
+    public ExportedProperties2() : this(0, null, null) {}
```

Comment:
I think we should also test the behaviour when no parameterless constructor is provided.

Reply:
The tests will pass either way, because the generators/analyzers don't check that the type has a parameterless constructor, but having no parameterless constructor will fail at runtime if the script is attached.

Reply:
Yeah. Now that I think about it, we don't check classes for a parameterless constructor even without using primary constructors. Maybe that's an analyzer we could add in the future. But the current test should be enough for this fix.